examples/model_compress/pruning/basic_pruners_torch.py

```diff
@@ -256,22 +206,82 @@ def main(args):
     flops, params, results = count_flops_params(model, dummy_input)
     print(f"FLOPs: {flops}, params: {params}")
 
-    print('start pruning...')
+    print(f'start {args.pruner} pruning...')
+    def trainer(model, optimizer, criterion, epoch, callback=None):
+        return train(args, model, device, train_loader, criterion, optimizer, epoch=epoch, callback=callback)
+        
+    pruner_cls = str2pruner[args.pruner]
+    
+    kw_args = {}
+    config_list = [{
+        'sparsity': args.sparsity,
+        'op_types': ['Conv2d']
+    }]
+
+    if args.dependency_aware:
+        dummy_input = get_dummy_input(args, device)
+        print('Enable the dependency_aware mode')
+        # note that, not all pruners support the dependency_aware mode
+        kw_args['dependency_aware'] = True
+        kw_args['dummy_input'] = dummy_input
+
+    if args.pruner == 'level':
+        config_list = [{
+            'sparsity': args.sparsity,
+            'op_types': ['default']
+        }]
+
+    else:
+        kw_args['trainer'] = trainer
+
+        if args.pruner not in ('l1filter', 'l2filter', 'fpgmfilter'):
+            # set only work for training aware pruners
+            kw_args['optimizer'] = optimizer
+            kw_args['criterion'] = criterion
+            kw_args['training_epochs'] = 5
+
+        # Reproduced result in paper 'PRUNING FILTERS FOR EFFICIENT CONVNETS',
+        # Conv_1, Conv_8, Conv_9, Conv_10, Conv_11, Conv_12 are pruned with 50% sparsity, as 'VGG-16-pruned-A'
+        if args.pruner == 'slim':
+            config_list = [{
+            'sparsity': args.sparsity,
+            'op_types': ['BatchNorm2d'],
+        }]
+        else:
+            config_list = [{
+            'sparsity': args.sparsity,
+            'op_types': ['Conv2d'],
+            'op_names': ['feature.0', 'feature.24', 'feature.27', 'feature.30', 'feature.34', 'feature.37']
```

Comment:
Why so specific?

Reply:
It aims to reproduce the result in the paper 'PRUNING FILTERS FOR EFFICIENT CONVNETS', in which prune `Conv_1, Conv_8, Conv_9, Conv_10, Conv_11, Conv_12`.